TypeScript program codegen: Requires types that other languages don't

<!-- Please provide a general summary of the issue, including what you're trying to accomplish. -->
This was found from compiling our program tests. Note that this test compiles fine in for `.NET` which is also strongly typed. That's why I think it's an issue with our TypeScript codegen. 
## Steps to reproduce
<!-- Provide a link to a live example or an unambiguous set of steps to reproduce this bug. Tell us what you expected to happen, and what happened instead. Please include code to reproduce, if relevant. -->

1. `cd pkg/codegen/internal/test/testdata/azure-native-pp`
2. `tsp --noEmit azure-native.ts`

Expected: A clean compile. 
Actual: 
```
azure-native.ts(4,67): error TS2345: Argument of type '{ routingRules: { routeConfiguration: { odataType: "#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration"; backendPool: { id: string; }; }; }[]; }' is not assignable to parameter of type 'FrontDoorArgs'.
  Property 'resourceGroupName' is missing in type '{ routingRules: { routeConfiguration: { odataType: "#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration"; backendPool: { id: string; }; }; }[]; }' but required in type 'FrontDoorArgs'.
azure-native.ts(12,60): error TS2345: Argument of type '{ deliveryPolicy: { rules: { actions: ({ name: "CacheExpiration"; parameters: { cacheBehavior: string; cacheDuration: string; cacheType: string; odataType: string; }; } | { name: "ModifyResponseHeader"; parameters: { ...; }; } | { ...; })[]; conditions: { ...; }[]; name: string; order: number; }[]; }; ... 6 more ......' is not assignable to parameter of type 'EndpointArgs'.
  Property 'origins' is missing in type '{ deliveryPolicy: { rules: { actions: ({ name: "CacheExpiration"; parameters: { cacheBehavior: string; cacheDuration: string; cacheType: string; odataType: string; }; } | { name: "ModifyResponseHeader"; parameters: { ...; }; } | { ...; })[]; conditions: { ...; }[]; name: string; order: number; }[]; }; ... 6 more ......' but required in type 'EndpointArgs'.
```

Also shows up in the `kuberntes-pod` test case.

--- a/pkg/codegen/testing/test/program_driver.go
+++ b/pkg/codegen/testing/test/program_driver.go
@@ -151,12 +151,10 @@ var PulumiPulumiProgramTests = []ProgramTest{
 	{
 		Directory:   "azure-native",
 		Description: "Azure Native",
-		SkipCompile: codegen.NewStringSet("go", "nodejs", "dotnet"),
+		SkipCompile: codegen.NewStringSet("go", "dotnet"),
 		// Blocked on go:
 		//   TODO[pulumi/pulumi#8073]
 		//   TODO[pulumi/pulumi#8074]
-		// Blocked on nodejs:
-		//   TODO[pulumi/pulumi#8075]
 	},
 	{
 		Directory:   "azure-native-v2-eventgrid",
@@ -188,12 +186,10 @@ var PulumiPulumiProgramTests = []ProgramTest{
 	{
 		Directory:   "kubernetes-pod",
 		Description: "K8s Pod",
-		SkipCompile: codegen.NewStringSet("go", "nodejs"),
+		SkipCompile: codegen.NewStringSet("go"),
 		// Blocked on go:
 		//   TODO[pulumi/pulumi#8073]
 		//   TODO[pulumi/pulumi#8074]
-		// Blocked on nodejs:
-		//   TODO[pulumi/pulumi#8075]
 	},
 	{
 		Directory:   "kubernetes-template",

